services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,69 @@
 package org.openkilda.functionaltests.spec.logging
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def tagName = "kilda-floodlight"
+        def result = elasticService.getLogs(new ElasticQueryBuilder().setTags(tagName).setTimeRange(300).build())
+        assert result?.hits?.total > 0
+
+        then: "There should be discovery messages"
+        result.hits.hits.any {hit -> hit.source.message.toLowerCase().contains("push discovery package via")}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        when: "A non-existent flow is requested"
+        def flowId = "nonexistentflowid" + System.currentTimeMillis()
+        try {
+            northbound.getFlow(flowId)
+        } catch (HttpClientErrorException e) {
+        }
+        and: "Rules on a switch are validated"
+        def switchId = topologyDefinition.switches[2].dpId
+        northbound.validateSwitchRules(switchId)
+
+        then: "Northbound, Storm and Topology Engine should log these actions within 1 minute"
+        def nbTag = "kilda-northbound"
+        def stormTag = "storm-worker_log"
+        def tpTag = "kilda-tpe"
+
+        boolean nbResult = false
+        boolean stormResult = false
+        boolean tpResult = false
+        int timeout = 60
```

Comment:
ditto.

Reply:
?

Reply:
it is also related to moving some stuff to constants.

Reply:
I see. It is pre-initialized variables rather than constants. Moved into setup section.